internal/services/network/edge_zone.go

```diff
@@ -27,3 +27,21 @@ func flattenEdgeZone(input *network.ExtendedLocation) string {
 	}
 	return edgezones.NormalizeNilable(input.Name)
 }
+
+func expandEdgeZoneModel(input string) *edgezones.Model {
+	normalized := edgezones.Normalize(input)
+	if normalized == "" {
+		return nil
+	}
+
+	return &edgezones.Model{
+		Name: normalized,
+	}
+}
+
+func flattenEdgeZoneModel(input *edgezones.Model) string {
+	if input == nil || input.Name == "" {
+		return ""
+	}
+	return edgezones.NormalizeNilable(&input.Name)
```

Comment:
we don't need to pass this as a pointer, we can instead use the regular `Normalize` method

```suggestion
	return edgezones.Normalize(input.Name)
```